cloud-functions/public/scripts/main.js

```diff
@@ -74,7 +74,7 @@ function loadMessages() {
         deleteMessage(change.doc.id);
       } else {
         var message = change.doc.data();
-        displayMessage(change.doc.id, message.timestamp, message.name,
+        displayMessage(change.doc.id, message.timestamp.seconds, message.name,
```

Comment:
I think method toMillis() would be more precise. `fromMillis()` option is also present for back conversion. Link to docs for more details: https://firebase.google.com/docs/reference/js/firebase.firestore.Timestamp
```suggestion
        displayMessage(change.doc.id, message.timestamp.toMillis(), message.name,
```

Reply:
yes, please use `toMillis()` instead, in case multiple messages come in during one second. Once you've done that, I'll merge this PR

Reply:
modified with requested changes